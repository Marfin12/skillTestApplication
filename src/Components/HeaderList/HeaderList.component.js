import * as React from 'react';
import { View } from 'react-native';
import { useSelector } from 'react-redux';

import styles from './HeaderList.styles';
import config from './HeaderList.config';
import TextField from '../TextField';

const HeaderList = (props) => {
  const getSelect = useSelector(({ themeReducer }) => themeReducer);
  const { theme } = getSelect;
  const { style, children } = props;

  return (
    <View style={[styles.container(theme), style]}>
      <TextField
        textStyle={styles.text(theme)}
      >
        {children}
      </TextField>
    </View>
  );
};

HeaderList.displayName = config.displayName;
HeaderList.defaultProps = config.defaultProps;

export default HeaderList;
