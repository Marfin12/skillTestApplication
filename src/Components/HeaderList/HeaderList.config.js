// @flow

import * as React from 'react';

import type { Props } from './HeaderList.types';

const defaultProps: Props = {
  style: {},
  children: <React.Fragment />
};

const displayName = 'HeaderList';

export default {
  defaultProps,
  displayName,
};
