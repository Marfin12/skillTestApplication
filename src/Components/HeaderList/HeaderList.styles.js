import colors from "../../Colors";

const styles = {
  container: darkTheme => {
    return {
      borderColor: darkTheme ? colors.darkHeavyGray : colors.lightHeavyGray,
      borderBottomWidth: 3,
    };
  },
  text: darkTheme => {
    return {
      fontSize: 20,
      fontWeight: 'bold',
      color: darkTheme ? colors.lightMediumGray : colors.blackPrimary,
    };
  },
};

export default styles;
