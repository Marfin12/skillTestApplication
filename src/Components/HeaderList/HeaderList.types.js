// @flow

import * as React from 'react';

import type {
  ViewStyleProp
} from 'react-native/Libraries/StyleSheet/StyleSheet';

export type Props = {|
  style: ViewStyleProp,
  children: React.Node
|};
