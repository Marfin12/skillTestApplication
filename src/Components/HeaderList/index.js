export {default} from './HeaderList.component';
