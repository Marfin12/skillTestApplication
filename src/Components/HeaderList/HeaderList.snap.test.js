import React from 'react';
import renderer from 'react-test-renderer';

import HeaderList from './HeaderList.component';
import config from './HeaderList.config';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => darkMode),
  };
});
jest.mock('../../Components/TextField', () => 'TextField');

afterEach(() => {
  jest.resetAllMocks();
});

describe('Header List Component Default Theme', () => {
  const ComponentWrapper = renderer
    .create(<HeaderList {...config.defaultProps} />)
    .toJSON();

  test('renders header list component light mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});

describe('Header List Component Dark Theme', () => {
  const ComponentWrapper = renderer
    .create(<HeaderList {...config.defaultProps} />)
    .toJSON();

  test('renders header list component dark mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});
