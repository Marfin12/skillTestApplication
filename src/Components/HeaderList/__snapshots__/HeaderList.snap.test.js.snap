// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header List Component Dark Theme renders header list component dark mode 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 3,
        "borderColor": "#222",
      },
      Object {},
    ]
  }
>
  <TextField
    textStyle={
      Object {
        "color": "#DDD",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  />
</View>
`;

exports[`Header List Component Default Theme renders header list component light mode 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 3,
        "borderColor": "#EEE",
      },
      Object {},
    ]
  }
>
  <TextField
    textStyle={
      Object {
        "color": "#000",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  />
</View>
`;
